clients/deltalake/core/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala

```diff
@@ -51,11 +52,17 @@ class NessieLogStore(sparkConf: SparkConf, hadoopConf: Configuration)
     NessieClient.builder().fromConfig(c => hadoopConf.get(c)).build()
   }
 
-  private def getOrCreate(): Reference = {
+  /**
+   * Keeps a mapping of reference name to current hash.
+   */
+  private val referenceMap: util.Map[String, Reference] = {
```

Comment:
why a `java` map and not a `scala` map?

Reply:
Just need a mutable map here - and using a Java `HashMap` felt to be the easiest way to go.